src/all/deviantart/src/eu/kanade/tachiyomi/extension/all/deviantart/DeviantArt.kt

```diff
@@ -76,28 +87,32 @@ class DeviantArt : HttpSource() {
 
     override fun mangaDetailsParse(response: Response): SManga {
         val document = response.asJsoup()
-        val subFolderGallery = document.selectFirst("#sub-folder-gallery")
-        val manga = SManga.create().apply {
-            // If manga is sub-gallery then use sub-gallery name, else use gallery name
-            title = subFolderGallery?.selectFirst("._2vMZg + ._2vMZg")?.text()?.substringBeforeLast(" ")
-                ?: subFolderGallery?.selectFirst("[aria-haspopup=listbox] > div")!!.ownText()
+        val gallery = document.selectFirst("#sub-folder-gallery")
+
+        // If manga is sub-gallery then use sub-gallery name, else use gallery name
+        val galleryName = gallery?.selectFirst("._2vMZg + ._2vMZg")?.text()?.substringBeforeLast(" ")
+            ?: gallery?.selectFirst("[aria-haspopup=listbox] > div")!!.ownText()
+        val artistInTitle = preferences.artistInTitle == ArtistInTitle.ON.name ||
+            preferences.artistInTitle == ArtistInTitle.ONLY_ALL.name && galleryName == "All"
```

Comment:
it isn't immediately clear what this means, so consider using meaningful name for the pref values to check against, or make an singleton object to group these (as you did with enum) if you want to